### PR TITLE
fix(infinitescroll): Move all ngOnInit operations to ngDoCheck

### PR DESCRIPTION
**Pull request by vikram-raj**
Fix - https://openshift.io/openshiftio/openshiftio/plan/detail/1728

### CODE REVIEW

File: src/app/infinitescroll/infinitescroll.directive.ts [hunk 1]
```diff
@@ -13,7 +13,8 @@ import {
     selector: '[almInfiniteScroll]',
     exportAs: 'almInfiniteScroll'
 })
-export class InfiniteScrollDirective implements OnInit {
+
```
Comment (joshuawilson) [inline src/app/infinitescroll/infinitescroll.directive.ts]: Why did you add a line between the class and the annotation? This may be minor but I just don't understand why.

File: src/app/infinitescroll/infinitescroll.directive.ts [hunk 2]
```diff
@@ -42,20 +42,22 @@ export class InfiniteScrollDirective implements OnInit {
...
-      let remainingHeight = this.element.scrollHeight - (this.element.offsetHeight + this.element.scrollTop);
+      let remainingHeight = this.element.scrollHeight
+        - (this.element.offsetHeight + this.element.scrollTop);
       let remainingElement = Math.ceil(remainingHeight / this.eachElementHeightInPx);
       if (remainingElement < this.fetchThreshold) {
         this.lastCheckedHeight = this.element.scrollHeight;
         this.fetchMore.emit();
       }
-    } else if (this.element.scrollHeight < this.previousScrollHeight) { // To check if the list is reloaded
+      // To check if the list is reloaded
```
Comment (joshuawilson) [inline src/app/infinitescroll/infinitescroll.directive.ts]: By moving the comment out of the block you are saying that it now belongs in the block above it. Is this your intention?

File: src/app/infinitescroll/infinitescroll.directive.ts [hunk 3]
```diff
@@ -22,13 +23,12 @@ export class InfiniteScrollDirective implements OnInit {
...
   previousScrollHeight = 0;
+  private element: HTMLElement = this.elementRef.nativeElement;
 
   constructor(private elementRef: ElementRef) {
   }
 
-  private element: HTMLElement = this.elementRef.nativeElement;
-
-  ngOnInit() {
+  ngAfterViewChecked() {
```
Comment (joshuawilson) [inline src/app/infinitescroll/infinitescroll.directive.ts]: The `ngAfterViewChecked` needs the `@ViewChild` to work properly. Why did you switch to this? What are you trying to do with it?

Comment (sanbornsen) [reply src/app/infinitescroll/infinitescroll.directive.ts]: I believe, ```ngAfterViewChecked``` is fired once the view is loaded finished mutation, and we need to check the height of the view at that point. The offsetHeight was giving 0 when the child height is auto because when ```ngOnInit``` called, children inside the view is not loaded.

File: src/app/infinitescroll/infinitescroll.directive.ts (hunk 1 above)
Comment (vikram-raj) [reply src/app/infinitescroll/infinitescroll.directive.ts]: Just to make it more readable.

File: src/app/infinitescroll/infinitescroll.directive.ts (hunk 2 above)
Comment (vikram-raj) [reply src/app/infinitescroll/infinitescroll.directive.ts]: I was getting linter error because it crosses 100 characters limit in a line. So that I move it to next line.

File: src/app/infinitescroll/infinitescroll.directive.ts (hunk 3 above)
Comment (joshuawilson) [reply src/app/infinitescroll/infinitescroll.directive.ts]: 'ngAfterViewChecked' is only for components, it is not meant for directives. Please find another way.

File: src/app/infinitescroll/infinitescroll.directive.ts (hunk 2 above)
Comment (joshuawilson) [reply src/app/infinitescroll/infinitescroll.directive.ts]: explaining it here won't help everyone else that reads it, you need to make the code right

File: src/app/infinitescroll/infinitescroll.directive.ts (hunk 1 above)
Comment (joshuawilson) [reply src/app/infinitescroll/infinitescroll.directive.ts]: this is not more readable as it violates the style standard

File: src/app/infinitescroll/infinitescroll.directive.ts (hunk 2 above)
Comment (joshuawilson) [reply src/app/infinitescroll/infinitescroll.directive.ts]: I misread this one, disregard.

File: src/app/infinitescroll/infinitescroll.directive.ts (hunk 3 above)
Comment (vikram-raj) [reply src/app/infinitescroll/infinitescroll.directive.ts]: I think another way to do this is with ``` ngDoCheck ```. Because ngDoCheck is for Components and as well for Directives.

**Comment by joshuawilson** [comment]
please clean up your commits, they are no longer accurate
